flink-applications.md

```diff
@@ -0,0 +1,203 @@
+
+<div class="row front-graphic">
+  <img src="{{ site.baseurl }}/img/function-state.png" width="350px" />
+</div>
+
+Application state is a first-class citizen in Flink. This is easy to see by looking at all the features that Flink provides in the context of state handling.
+
+* **Keyed State** and **Operator State**: Flink provides two types of managed state, which are accessed in different contexts. Keyed state is always accessed in the context of a key, operator state is accessed in the context of an operator instance.
+* **Multiple State Primitives**: Flink provides state primitives for different data structures, such as atomic values, lists, or maps. Developers can choose the state primitive that is most efficient based on the access pattern of the function.
+* **Pluggable State Backends**: Application state is managed in and checkpointed by a pluggable state backend. Flink features different state backends that store state in memory or in RocksDB, an efficient embedded on-disk data store. Also custom state backends can be plugged in.
+* **Exactly-once state consistency**: Flink's checkpointing and recovery algorithms guarantee the consistency of application state in case of a failure. Hence, failures are transparently handled and do not affect the correctness of an application.
+* **Very Large State**: Flink is able to maintain application state of several terabytes in size due to its asynchronous and incremental checkpoint algorithm.
+* **Scalable Applications**: Flink supports scaling of stateful applications by redistributing the state to more or fewer workers.
+
+### Time
+
+Time is another important ingredient of streaming applications. Most event streams have inherent time semantics because each event is produced at a speicific point in time. Moreover, many common stream computations are based on time, such as windows aggregations, sessionization, pattern detection, and time-based joins. An important aspect of stream processing is how an application measures time, i.e., the difference of event-time and processing-time.
+
+Flink provides a rich set of time-related features.
+
+* **Event-time Mode**: Applications that process streams with event-time semantics compute results based on timestamps of the events. Thereby, event-time processing allows for accurate and consistent results regardless whether recorded or real-time events are processed.
+* **Watermark Support**: Flink employs watermarks to reason about time in event-time applications. Watermarks are also a flexible mechanism to trade-off the latency and completeness of results.
+* **Late Data Handling**: When processing streams in event-time mode with watermarks, it can happen that a computation has been completed before all associated events have arrived. Such events are called late events. Flink features multiple options to handle late events, such as rerouting them via side outputs and updating previously completed results.
```

Comment:
add link to side output docs?

Reply:
The side output docs don't discuss the late data use case.

Reply:
I meant just as an additional reference.